### DISCUSSION TITLE
Custom sign in page for Keycloak provider

### DISCUSSION THREAD

**Discussion by Nmohyieddine** [nextauthjs/next-auth#10860] [Help]
Hey everyone, I'm currently exploring integrating NextAuth with the Keycloak provider. My goal is to leverage Keycloak IAM while employing a custom sign-in form. Upon researching, I discovered that to avoid redirection to the default sign-in page of the provider, I can override it using the `pages` option in the `AuthOptions` object. However, I'm now facing difficulty in ensuring that my application captures sign-in information and passes it to Keycloak. I'm wondering if NextAuth offers functionality for this or if I should utilize a credentials provider and then employ the Keycloak-JS adapter for this specific use case.

Here is my route.tsx file :

```typescript
import { AuthOptions } from "next-auth";
import NextAuth from "next-auth/next";
import KeycloakProvider from "next-auth/providers/keycloak";

export const authOptions: AuthOptions = {
  providers: [
    KeycloakProvider({
      clientId: process.env.KEYCLOAK_ID as string,
      clientSecret: process.env.KEYCLOAK_SECRET as string,
      issuer: process.env.KEYCLOAK_ISSUER,
    }),

  ],
  pages: {
    signIn: '/signIn',
    signOut: '/auth/signout',
    error: '/auth/error', // Error code passed in query string as ?error=
    verifyRequest: '/auth/verify-request', // (used for check email message)
    newUser: '/auth/new-user' // New users will be directed here on first sign in (leave the property out if not of interest)
  }
}

const handler = NextAuth(authOptions);


export { handler as GET, handler as POST }
```
and the example of the signIn form i would like to use :

![Screen Shot 2024-05-09 at 11 20 32 AM](https://github.com/nextauthjs/next-auth/assets/73003863/66ad4079-2259-4407-a37b-ed9a204761e0)

**Comment by devcshort** [comment]
I'm also interested in this. @Nmohyieddine please let me know what you end up going with as this is something I'm going to have to tackle in the near future as well. Would be nice to have some insight on it. Thanks!

**Comment by benmarte** [comment]
I've used this in the past to create a keycloak theme and it works great: https://www.keycloakify.dev/
